Sources/MongoSwift/BSON/DocumentIterator.swift

```diff
@@ -161,16 +166,34 @@ public class DocumentIterator: IteratorProtocol {
     }
 
     /// Internal helper function for explicitly accessing the `bson_iter_t` as an unsafe pointer
+#if compiler(>=5.0)
     internal func withBSONIterPointer<Result>(_ body: (BSONIterPointer) throws -> Result) rethrows -> Result {
-      return try withUnsafePointer(to: self.iter, body)
+        return try withUnsafePointer(to: self.iter) { iterPtr in
+            try body(OpaquePointer(iterPtr))
```

Comment:
In this cast and the one below we can use the typealiases above, unless you meant for the opaque-ness to be explicit.